### PR TITLE
gnrc_ipv6_nib: add exception in address resolution for SLIP

### PR DESCRIPTION
**Pull request by miri64**
Linux doesn't answer to NS over SLIP and since there are no link-layer addresses in SLIP, address resolution doesn't make much sense anyway, this PR circumvents it for SLIP interfaces. If the interface is provided to the address resolution (which it is for link-local or if a route is configured, which is the case for a border router where SLIP is usually used) the returned neighbor cache entry is just set to that interface with link-layer address 0.

### CODE REVIEW

**Comment by bergzand** [comment]
Ignoring any instabilities, ping works! I suspect the issue to be somewhere in the uart code of the nucleo. A value of `0xff` sometimes lead the frame confusing the tunslip6 application. On the nucleo-side of the uart it is also easily possible to break connectivity, again suspected to happen in the UART code.

To get things working:
1. disconnect the nucleo-TX, host side RX wire.
2. start tunslip6 application (using aaaa::1/64 in this example)
3. add aaaa::2/64 on the nucleo side to the interface
4. start a ping6 on the host to aaaa::2
5. while keeping the ping6 running, connect the TX wire again.

This way the tunslip application doesn't receive the `0xff` byte and the ping6 on the host side should start reporting responses (around 30ms at 115200 baud). pinging from the node to the host also works, with responses around 20ms.

**TL;DR**: it unstable, but it works, and this PR is not to blame for the instability.

**Comment by miri64** [comment]
@bergzand can you open an issue regarding your suspicions? I'm not really that much into the periph code of any platform and this way we can notify people about this, that do actually know about that code ;-).

**Comment by miri64** [comment]
(fixed some wording in the comment in my latest commit)

**Comment by miri64** [comment]
And rebased to make testing easier.

File: sys/net/gnrc/network_layer/ipv6/nib/nib.c [hunk 1]
```diff
@@ -626,6 +626,15 @@ static bool _resolve_addr(const ipv6_addr_t *dst, gnrc_netif2_t *netif,
                           _nib_onl_entry_t *entry)
 {
     bool res = false;
+    if ((netif != NULL) && (netif->device_type == NETDEV_TYPE_SLIP)) {
+        /* XXX: Linux doesn't do neighbor discovery for SLIP so no use sending
+         * NS and since SLIP doesn't has link-layer addresses anyway we can just
```
Comment (bergzand) [inline sys/net/gnrc/network_layer/ipv6/nib/nib.c]: has -> have, and a comma after "anyway" (I think).

Comment (miri64) [reply sys/net/gnrc/network_layer/ipv6/nib/nib.c]: Fixed

**Comment by bergzand** [comment]
Ack, please squash

**Comment by miri64** [comment]
Squashed

> @bergzand can you open an issue regarding your suspicions? I'm not really that much into the periph code of any platform and this way we can notify people about this, that do actually know about that code ;-).

Please remember to do that.

**Comment by bergzand** [comment]
> Please remember to do that.

Thanks for the reminder, #8045 :)

**Comment by miri64** [comment]
Errors on Murdock are all related to #7414. @bergzand would you like to have the honor of hitting merge ;-)?

**Comment by bergzand** [comment]
I'm afraid I don't have the permissions to do that 😢
permission_denied

**Comment by miri64** [comment]
Weird... can you do to master? Might be that I misconfigured something for `gnrc_netif2_integration/master`.

**Comment by bergzand** [comment]
Yeah, master works (#7977). I don't really mind for this case, but it might explain why I never merged anything on this branch :)

**Comment by miri64** [comment]
Now you should be able to :-)

**Comment by miri64** [comment]
(as should any non-admin maintainer)

**Comment by bergzand** [comment]
:tada:
